cranelift/codegen/src/isa/s390x/inst.isle

```diff
@@ -3424,17 +3427,17 @@
             ;; Emit initial load followed by compare-and-swap loop.
             (_ Unit (emit_load (ty_ext32 ty) (casloop_val_reg) aligned_mem))
             (_ Unit (emit_loop ib (intcc_as_cond (IntCC.NotEqual)))))
-        result))
+        (preg_gpr_0)))
```

Comment:
This a bit iffy, in that it's not guaranteed (at least, that wasn't the intention) that `result` *must* be GPR 0 here.  I think in practice it (nearly?) always is, but it still would be better to at least verify.

Reply:
I'll verify that it's always used with gpr 0, and add a comment explaining why we return it there if that's the case.